packaging/setup/plugins/ovirt-engine-setup/ovirt-engine-keycloak/apache/keycloakproxy.py

```diff
@@ -0,0 +1,75 @@
+#
+# ovirt-engine-setup -- ovirt engine setup
+#
+# Copyright oVirt Authors
+# SPDX-License-Identifier: Apache-2.0
+#
+#
+
+"""keycloak httpd proxy plugin."""
+
+import gettext
+
+from otopi import constants as otopicons
+from otopi import filetransaction
+from otopi import plugin
+from otopi import util
+from ovirt_engine import util as outil
+from ovirt_engine_setup import constants as osetupcons
+from ovirt_engine_setup.engine import constants as oenginecons
+from ovirt_engine_setup.engine_common import constants as oengcommcons
+from ovirt_engine_setup.keycloak import constants as okkcons
+
+
+def _(m):
+    return gettext.dgettext(message=m, domain='ovirt-engine-keycloak')
+
+
+@util.export
+class Plugin(plugin.PluginBase):
+    """keycloak httpd proxy plugin."""
+
+    def __init__(self, context):
+        super(Plugin, self).__init__(context=context)
+
+    @plugin.event(
+        stage=plugin.Stages.STAGE_INIT,
+    )
+    def _init(self):
+        self.environment.setdefault(
+            okkcons.ApacheEnv.HTTPD_CONF_OVIRT_ENGINE_KEYCLOAK,
+            okkcons.FileLocations.HTTPD_CONF_OVIRT_ENGINE_KEYCLOAK
+        )
+
+    @plugin.event(
+        stage=plugin.Stages.STAGE_MISC,
+        condition=lambda self: self.environment[
```

Comment:
indent

No other conditions? If engine is enabled, keycloak is mandatory?

Reply:
Keycloak is the default of all new installation of oVirt 4.5. For oVirt upgrade from 4.4 to 4.5 we plan to leave current AAA configuration as is, admins will need to migrate from AAA to Keycloak manually.

For RHV we don't plan to package Keycloak, administrators can use external RH SSO setup as it's possible in 4.4

Reply:
For now there is only this configuration option, however, I expect more customization parameters to come.

Reply:
Please create an issue for this, if not handling here.

Reply:
Issue #9 created